docs/references/nextjs/clerk-middleware.mdx

```diff
@@ -428,3 +435,86 @@ export default clerkMiddleware(
 
 > [!NOTE]
 > Dynamic keys are not accessible on the client-side.
+
+### OrganizationSyncOptions
+
+The `organizationSyncOptions` property on the <code>[clerkMiddleware()](#clerk-middleware-options)</code> options
+object has the type `OrganizationSyncOptions`.
+
+This defines the options for automatically activating an organization or the [personal account](/glossary#personal-account)
+based on the URL. This is useful for applications that require organization slugs or IDs to appear in URL paths.
+
+If a mismatch between the active organization on the session (e.g. as reported by
+<code>[auth()](/docs/references/nextjs/auth)</code>) and the organization as indicated by the URL is detected, an attempt
+to activate the given organization will be made.
+
+`OrganizationSyncOptions` has the following properties:
+
+<Properties>
+  - `organizationPatterns`
+  - <code>[Pattern](#pattern)\[]</code>
+
+  URL patterns that are organization-specific and contain an organization ID or slug as a path parameter. If a request
+  matches this path, the organization identifier will be extracted and activated before rendering.
+
+  If the route also matches the personalAccountPatterns, this takes precedence.
+
+  Patterns must have a path parameter named either `:id` (matches a clerk organization ID) or `:slug` (matches a clerk
+  organization slug).
+
+  > [!WARNING]
+  > If the organization cannot be activated, either because it does not exist or the user lacks access,
+  > no change will be made to the previously active organization. Components must detect this case and respond with an
+  > appropriate error (e. g., `notFound()`, an
+  > [organization switcher](/docs/components/organization/organization-switcher), etc.).
```

Comment:
🔧 This sentence and examples in parentheses was confusing to me, because responding by rendering the org switcher doesn't seem to fall into the category of an appropriate _error_

Reply:
73ce834fcd8913bc2cd41ef67ec313a4ac7a2f16

How does that feel?